db/migrations/0018_add_lti_usage.py

```diff
@@ -0,0 +1,30 @@
+'''
+Migration for LTI Usage tables
+'''
+
+from yoyo import step
+
+steps = [
+    step('''
+        CREATE TABLE IF NOT EXISTS lti_placement (
+            id INTEGER NOT NULL,
+            account_id INTEGER NOT NULL,
+            course_name VARCHAR(200),
+            placement_type_id INTEGER NOT NULL,
+            PRIMARY KEY (id, placement_type_id)
+        ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4;
+    '''),
+    step('''
+        CREATE TABLE IF NOT EXISTS lti_zoom_meeting (
+            id BIGINT NOT NULL,
+            course_id INTEGER NOT NULL,
+            meeting_id VARCHAR(100) NOT NULL,
+            host_id VARCHAR(100) NOT NULL,
+            topic VARCHAR(255) NOT NULL,
+            start_time DATETIME NOT NULL,
+            status INTEGER,
+            PRIMARY KEY (id),
+            FOREIGN KEY (course_id) REFERENCES lti_placement(id) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
Is using `course_id` as the foreign key going to work if or when we have multiple LTI types in `lti_placement`? Seems like this would create some inaccurate joins. Is it possible to use the compound key?

Reply:
Maybe `id` should stay `course_id`? I feel that's going to become confusing, since that's the convention (or at least my convention) for auto-incremented PKs, and alone it does not uniquely identify the `lit_placement` entity.

Reply:
My idea was that `id` in `lti_placement` would just be `course_id`. This + `placement_type_id` is unique. However you're right, without a separate auto incremented id you're right it would have to have a foreign composite key once more were added. However looking at the documentation I can't just have something like

`FOREIGN KEY (course_id, 1) REFERENCES lti_placement(id, placement_type_id) ON DELETE CASCADE ON UPDATE CASCADE`

It actually would need to be a column in `lti_zoom_meeting` to work. I switched this around only because I couldn't figure out an *easy* way of keeping the auto incremented id linked since the process.

- Pandas `to_sql` inserts all of the values into `lti_placement`
- Pandas `to_sql` Then inserts all of the values into `lti_zoom_meeting`

I'd probably have to keep a map in the code, or do a lookup against the dataframe? I guess that would work but it's more code complexity. I also figured it was better to leave the auto increment to pandas rather than SQL but that's going to cause problems when/if we upsert instead of deleting and inserting. 

1) Add a column in LTI zoom meeting (`with DEFAULT 1`) and do a composite key
2) Add an auto increment field to `lti_placement`, have course_id separate. Create some process to linkup the `lti_zoom_meeting` id to this auto increment id. I tried to think of the best way to do this and didn't quickly while still using `to_sql`. I'll think about this, maybe I can use a merge. 

Is this something you're already doing?

Reply:
Yeah, you've described the issue well. I wish there was some kind of unique identifier created by Canvas for an installed LTI app. Are you aware of any?

I am doing something like this for `job_run` and `data_source_status`, but it's pretty basic. It does things one at a time, and those never will need to be updated:

https://github.com/tl-its-umich-edu/course-inventory/blob/163704da5037b81998e858f9c23225ba8fe7a9e0/run_jobs.py#L45

Reply:
In the URL for an LTI, I am seeing an external tool ID: `courses/{course_id}/external_tools/{external_tool_id}`. I'm guessing that's for LTI app, not the placement. Maybe we could cheat and just smush these together?

Reply:
Yeah, that's a good idea! That will require some code refactor but I think it could work for both cases. I had an idea for delete/insert that would work if we generated the id's in the code but not if we switched to upsert so using something like this makes more sense.

Reply:
Okay, so the external_tool_id isn't actually unique per course. It's the same for Zoom in every course. course + this id is unique but that really doesn't matter since this is static.

I think I'm going to have to auto increment in this code and optimize it later.

Reply:
I resolved this on the latest commit. I had to force push as there were conflicts with another recent change.